stograde/student/markdownify/process_file.py

```diff
@@ -99,9 +100,10 @@ def test_file(filename, *, spec, results, options, cwd, supporting_dir, interact
     return True
 
 
-def process_file(filename, *, steps, options, spec, cwd, supporting_dir, interact, basedir, spec_id,
-                 skip_web_compile):
+def process_file(filename, *, steps, tests, options, spec, cwd, supporting_dir, interact, basedir,
+                 spec_id, skip_web_compile):
     steps = steps if isinstance(steps, Iterable) else [steps]
+    tests = tests if isinstance(tests, Iterable) else [tests]
```

Comment:
Unfortunately, strings are iterable.

How about flipping the sides?

```suggestion
    tests = [tests] if isinstance(tests, str) else tests
```

Reply:
Should we do this for `steps` too? Though in theory both should already be arrays because of how they are cached, right? Or am I missing something there

Reply:
As far as I can tell, lines [97-99](https://github.com/StoDevX/stograde/blob/aa6115bfb60cc713525f8f9a8b36d162e9ef109c/stograde/specs/cache.py#L97) and [113](https://github.com/StoDevX/stograde/blob/aa6115bfb60cc713525f8f9a8b36d162e9ef109c/stograde/specs/cache.py#L113) in `cache.py` should take care of this, right?

Reply:
looks good to me